src/main/java/com/redhat/cloud/notifications/processors/camel/CamelTypeProcessor.java

```diff
@@ -0,0 +1,129 @@
+    }
+
+    public Uni<NotificationHistory> process(Notification item) {
+        processedCount.increment();
+        Endpoint endpoint = item.getEndpoint();
+        CamelProperties properties = (CamelProperties) endpoint.getProperties();
+
+        Map<String, String> metaData = new HashMap<>();
+
+        metaData.put("trustAll", String.valueOf(properties.getDisableSslVerification()));
+
+        metaData.put("url", properties.getUrl());
+        metaData.put("type", properties.getSubtype());
+
+        if (properties.getSecretToken() != null && !properties.getSecretToken().isBlank()) {
+            metaData.put(TOKEN_HEADER, properties.getSecretToken());
+        }
+
+        BasicAuthentication basicAuthentication = properties.getBasicAuthentication();
+        if (basicAuthentication != null) {
+            StringBuilder sb = new StringBuilder(basicAuthentication.getUsername());
+            sb.append(":");
+            sb.append(basicAuthentication.getPassword());
+            String b64 = Base64.getEncoder().encodeToString(sb.toString().getBytes(StandardCharsets.UTF_8));
+            metaData.put("basicAuth", b64);
+        }
+
+        metaData.put("extras", new MapConverter().convertToDatabaseColumn(properties.getExtras()));
+
+        Uni<JsonObject> payload = transformer.transform(item.getAction());
+        return callCamel(item, metaData, payload);
+    }
+
+    public Uni<NotificationHistory> callCamel(Notification item, Map<String, String> meta, Uni<JsonObject> payload) {
+
+        final long startTime = System.currentTimeMillis();
+        final Map<String, Object> tmp = new HashMap<>();
+        UUID historyId = UUID.randomUUID();
+        meta.put("historyId", historyId.toString());
+
+        Uni<NotificationHistory> historyUni = payload.onItem()
+                .transform(json -> {
+                    tmp.put("meta", meta);
+                    tmp.put("payload", json);
+                    return tmp;
+                })
+                .onItem().transformToUni(json -> reallyCallCamel(json)
+                        .onItem().transform(resp -> {
+                            final long endTime = System.currentTimeMillis();
+                            // We only create a basic stub. The FromCamel filler will update it later
+                            NotificationHistory history = getHistoryStub(item, endTime - startTime, historyId);
+                            return history;
+                        })
+                );
+
+        return historyUni;
+    }
+
+    public Uni<Boolean> reallyCallCamel(Map<String, Object> body) {
+
+        JsonObject jo = JsonObject.mapFrom(body);
+
+        TracingMetadata tracingMetadata = TracingMetadata.withPrevious(Context.current());
+        Message<String> msg = Message.of(jo.encode()); //, Metadata.of(tracingMetadata));
+
+        return Uni.createFrom().item(() -> {
+            emitter.send(jo.encode()); // TODO use msg
```

Comment:
Should this TODO be fixed before the merge? `tracingMetadata` and `msg` are never used.

Reply:
I wanted to get tracing in, but Quarkus is/was not yet ready for this, so those are left-overs. I guess we can do both ways.